[Bug]: test_gateway_service systemd refresh tests fail in non-systemd environments after user D-Bus preflight

## Bug Description
`tests/hermes_cli/test_gateway_service.py` has refresh tests that now fail in non-systemd environments because `systemd_start()` / `systemd_restart()` call `_preflight_user_systemd()` before the tests' mocked `subprocess.run` path.

This appears to be a test regression introduced after adding user D-Bus preflight in `hermes_cli/gateway.py`.

## Steps to Reproduce
1. Fresh checkout of `main`.
2. Create env and install deps:
   ```bash
   uv venv .venv --python 3.11
   source .venv/bin/activate
   uv pip install -e ".[all,dev]"
   ```
3. Run targeted tests:
   ```bash
   pytest -q -n 0 \
     tests/hermes_cli/test_gateway_service.py::TestSystemdServiceRefresh::test_systemd_start_refreshes_outdated_unit \
     tests/hermes_cli/test_gateway_service.py::TestSystemdServiceRefresh::test_systemd_restart_refreshes_outdated_unit
   ```

## Expected Behavior
These unit tests should stay hermetic and pass by exercising mocked systemctl/unit-refresh logic without requiring a real user D-Bus/systemd session.

## Actual Behavior
Both tests fail before assertions with:

```text
hermes_cli.gateway.UserSystemdUnavailableError:
User D-Bus session is not available (...)
```

## Affected Component
- CLI (gateway service management test path)
- Test suite

## Messaging Platform (if gateway-related)
- N/A (CLI/test path)

## Debug Report
Not applicable for this unit-test regression report.

## Operating System
macOS (arm64)

## Python Version
3.11.15

## Hermes Version
Source checkout of `main` at commit `a0d8dd7b` (`pyproject.toml` version `0.10.0`).

## Additional Logs / Traceback (optional)
Minimal failure signal:

```text
FAILED tests/hermes_cli/test_gateway_service.py::TestSystemdServiceRefresh::test_systemd_start_refreshes_outdated_unit
FAILED tests/hermes_cli/test_gateway_service.py::TestSystemdServiceRefresh::test_systemd_restart_refreshes_outdated_unit
E hermes_cli.gateway.UserSystemdUnavailableError: User D-Bus session is not available (...)
```

## Root Cause Analysis (optional)
Likely regression from commit `d45c738a` (`fix(gateway): preflight user D-Bus before systemctl --user start (#14531)`).

`systemd_start()` and `systemd_restart()` now call `_preflight_user_systemd()`:
- `hermes_cli/gateway.py:1826`
- `hermes_cli/gateway.py:1847`

The affected tests (`tests/hermes_cli/test_gateway_service.py`, lines ~41-87) stub `subprocess.run` but do not stub `_preflight_user_systemd()`, so they now fail on non-systemd/non-user-DBus environments before hitting their intended assertions.

## Proposed Fix (optional)
Keep the runtime preflight behavior, but patch tests to be explicit about this dependency, e.g. monkeypatch `_preflight_user_systemd` to a no-op in the affected test cases (or a fixture used by this class), then keep existing assertions on daemon-reload/start/restart command ordering.

## Are you willing to submit a PR for this?
Open to submit a PR after maintainer confirmation on preferred test strategy.

--- a/tests/hermes_cli/test_gateway_service.py
+++ b/tests/hermes_cli/test_gateway_service.py
@@ -68,6 +68,10 @@ class TestSystemdServiceRefresh:
 
         monkeypatch.setattr(gateway_cli, "get_systemd_unit_path", lambda system=False: unit_path)
         monkeypatch.setattr(gateway_cli, "generate_systemd_unit", lambda system=False, run_as_user=None: "new unit\n")
+        # Bypass systemd availability checks — this test targets unit-file
+        # refresh logic, not D-Bus reachability (fails on macOS/WSL/Docker).
+        monkeypatch.setattr(gateway_cli, "_preflight_user_systemd", lambda **kw: None)
+        monkeypatch.setattr(gateway_cli, "_select_systemd_scope", lambda system=False: False)
 
         calls = []
 
@@ -91,6 +95,10 @@ class TestSystemdServiceRefresh:
 
         monkeypatch.setattr(gateway_cli, "get_systemd_unit_path", lambda system=False: unit_path)
         monkeypatch.setattr(gateway_cli, "generate_systemd_unit", lambda system=False, run_as_user=None: "new unit\n")
+        # Bypass systemd availability checks — this test targets unit-file
+        # refresh logic, not D-Bus reachability (fails on macOS/WSL/Docker).
+        monkeypatch.setattr(gateway_cli, "_preflight_user_systemd", lambda **kw: None)
+        monkeypatch.setattr(gateway_cli, "_select_systemd_scope", lambda system=False: False)
 
         calls = []
         monkeypatch.setattr("gateway.status.get_running_pid", lambda: None)

